### ISSUE TITLE
Why is the training speed too slow and is the performance so weird on 4 Titan X GPU ?

### ISSUE THREAD

**Issue by bruinxiong** [dmlc/mxnet#5074]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System: Ubuntu 14.04.5

Compiler:  gcc 4.8.4

Package used (Python/R/Scala/Julia): python

MXNet version: 

Or if installed from source:  installed by git clone https://github.com/dmlc/mxnet.git ~/mxnet --recursive

MXNet commit hash (`git rev-parse HEAD`):

If you are using python package, please provide

Python version and distribution: python 2.7.13 anaconda 4.3.0

If you are using R package, please provide

R `sessionInfo()`:

## Error Message:
Please paste the full error message, including stack trace.

## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.

## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1.
2.
3.

## What have you tried to solve it?

1.
2.
3.
Build environment
nvidia 375.26, cuda 8.0, gcc 4.8.4, ubundu 14.04.5, cudnn 5.1

When I use 4 pascal GPUs ( Titan X) to retrain ResNet 50 from model_load_epoch=90 on imagenet'12 dataset. The speed is very slow. Furthermore, the accuracy is also wrong. The following picture shows two procedures. And there are two weird accuracy. ( MXNET is the latest version )

I don't know why. Before change to pascal GPU, I use 4 M40 GPU have no issue ( Old MXNET version, please see the second picture ) .

![image](https://cloud.githubusercontent.com/assets/3366247/23129516/f19b3afe-f7bd-11e6-9f29-e089c7888c68.png)

Obviously, this is an abnormal phenomenon.

Normally, the results should be shown following as 

![image](https://cloud.githubusercontent.com/assets/3366247/23129668/89631b9a-f7be-11e6-8c94-3951d057b8f8.png)

**Comment by piiswrong** [comment]
@ap-hynninen Any idea?

**Comment by ap-hynninen** [comment]
I've trained resnet on Titan X Pascal and didn't see a slow down. How does the code perform when you run it on a single Titan X Pascal ?

**Comment by bruinxiong** [comment]
@ap-hynninen  Sorry for late reply. We need some time to test. 
We uses pre-installed mxnet example, first, we train cifar 10 with single GPU,   with python train_cifar10.py --gpus 7. **There is no problem.** The training procedure is listed as following

➜  image-classification git:(master) ✗ sudo python train_cifar10.py --gpus 7
INFO:root:start with arguments Namespace(batch_size=128, benchmark=0, data_nthreads=4, data_train='data/cifar10_train.rec', data_val='data/cifar10_val.rec', disp_batches=20, gpus='7', image_shape='3,28,28
', kv_store='device', load_epoch=None, lr=0.05, lr_factor=0.1, lr_step_epochs='200,250', max_random_aspect_ratio=0, max_random_h=36, max_random_l=50, max_random_rotate_angle=0, max_random_s=50, max_random
_scale=1, max_random_shear_ratio=0, min_random_scale=1, model_prefix=None, mom=0.9, monitor=0, network='resnet', num_classes=10, num_epochs=300, num_examples=50000, num_layers=110, optimizer='sgd', pad_si
ze=4, random_crop=1, random_mirror=1, rgb_mean='123.68,116.779,103.939', test_io=0, top_k=0, wd=0.0001)
[10:23:37] src/io/iter_image_recordio.cc:221: ImageRecordIOParser: data/cifar10_train.rec, use 4 threads for decoding..
[10:23:37] src/io/iter_image_recordio.cc:221: ImageRecordIOParser: data/cifar10_val.rec, use 4 threads for decoding..
[10:24:34] src/operator/./cudnn_convolution-inl.h:55: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to 
disable)
INFO:root:Epoch[0] Batch [20]   Speed: 1322.98 samples/sec      Train-accuracy=0.152530
INFO:root:Epoch[0] Batch [40]   Speed: 1388.75 samples/sec      Train-accuracy=0.171875
INFO:root:Epoch[0] Batch [60]   Speed: 1375.13 samples/sec      Train-accuracy=0.221094
INFO:root:Epoch[0] Batch [80]   Speed: 1383.92 samples/sec      Train-accuracy=0.225000
INFO:root:Epoch[0] Batch [100]  Speed: 1390.25 samples/sec      Train-accuracy=0.246484
INFO:root:Epoch[0] Batch [120]  Speed: 1363.02 samples/sec      Train-accuracy=0.260937
INFO:root:Epoch[0] Batch [140]  Speed: 1377.46 samples/sec      Train-accuracy=0.262109
INFO:root:Epoch[0] Batch [160]  Speed: 1386.32 samples/sec      Train-accuracy=0.291406
INFO:root:Epoch[0] Batch [180]  Speed: 1402.37 samples/sec      Train-accuracy=0.289062
INFO:root:Epoch[0] Batch [200]  Speed: 1375.69 samples/sec      Train-accuracy=0.299219
INFO:root:Epoch[0] Batch [220]  Speed: 1389.71 samples/sec      Train-accuracy=0.313672
INFO:root:Epoch[0] Batch [240]  Speed: 1398.73 samples/sec      Train-accuracy=0.310156
INFO:root:Epoch[0] Batch [260]  Speed: 1374.34 samples/sec      Train-accuracy=0.317578
INFO:root:Epoch[0] Batch [280]  Speed: 1391.53 samples/sec      Train-accuracy=0.332031
INFO:root:Epoch[0] Batch [300]  Speed: 1382.85 samples/sec      Train-accuracy=0.335547
INFO:root:Epoch[0] Batch [320]  Speed: 1353.69 samples/sec      Train-accuracy=0.358203
INFO:root:Epoch[0] Batch [340]  Speed: 1371.99 samples/sec      Train-accuracy=0.344922
INFO:root:Epoch[0] Batch [360]  Speed: 1395.19 samples/sec      Train-accuracy=0.364844

**However, we train with 2 GPUs, the training speed is extremely slow:**

➜  image-classification git:(master) ✗ sudo python train_cifar10.py --gpus 6,7
INFO:root:start with arguments Namespace(batch_size=128, benchmark=0, data_nthreads=4, data_train='data/cifar10_train.rec', data_val='data/cifar10_val.rec', disp_batches=20, gpus='6,7', image_shape='3,28,28', kv_store='device', load_epoch=None, lr=0.05, lr_factor=0.1, lr_step_epochs='200,250', max_random_aspect_ratio=0, max_random_h=36, max_random_l=50, max_random_rotate_angle=0, max_random_s=50, max_random_scale=1, max_random_shear_ratio=0, min_random_scale=1, model_prefix=None, mom=0.9, monitor=0, network='resnet', num_classes=10, num_epochs=300, num_examples=50000, num_layers=110, optimizer='sgd', pad_size=4, random_crop=1, random_mirror=1, rgb_mean='123.68,116.779,103.939', test_io=0, top_k=0, wd=0.0001)
[10:25:19] src/io/iter_image_recordio.cc:221: ImageRecordIOParser: data/cifar10_train.rec, use 4 threads for decoding..
[10:25:19] src/io/iter_image_recordio.cc:221: ImageRecordIOParser: data/cifar10_val.rec, use 4 threads for decoding..
[10:25:22] src/operator/./cudnn_convolution-inl.h:55: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
INFO:root:Epoch[0] Batch [20]   Speed: 7.56 samples/sec Train-accuracy=0.101562 


So, please help us to solve this problem

Thank you very much.

**Comment by ap-hynninen** [comment]
I can't reproduce this. I think it's something to do with your particular setup. I suggest just trying things like
running in benchmark mode ( --benchmark 1), run it through a profiler that shows you where the CPU is spending it's time. You can profile the GPU with:
"nvprof python train_cifar10.py --gpus 6,7"

Look for the [CUDA memcpy HtoD] and [CUDA memcpy DtoH] tags, they tell you how much is spent in copying data between CPU and GPU.

Btw. Why are you running this under sudo?

**Comment by bruinxiong** [comment]
@ap-hynninen 
Sorry for late reply. We final find the reason by consulting our server vendor. This is a bug in how the Intel V4 Quick Path interconnect interfaces with PCI-E. Finally, we disable the ACS control in Bios->Advanced->Chipset Configuration->North Bridge->IO Configuration->Intel VT for Directed I/O(VT-d)->ACS Control.

But we have another issue about speed (xxx samples/sec) in training procedure. Please see this link https://github.com/dmlc/mxnet/issues/5365
